components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/UnsupportedAddonsAdapter.kt

```diff
@@ -42,28 +44,40 @@ class UnsupportedAddonsAdapter(
                 addon.id
             }
 
+        bindRemoveButton(holder, addon)
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun bindRemoveButton(holder: UnsupportedAddonViewHolder, addon: Addon) {
+        holder.removeButton.isEnabled = !pendingUninstall
+        if (pendingUninstall) {
+            return
+        }
+
         holder.removeButton.setOnClickListener {
-            holder.removeButton.visibility = View.GONE
+            // notify all visible view holders to disable removeButton
+            pendingUninstall = true
+            notifyDataSetChanged()
             addonManager.uninstallAddon(addon,
-                onSuccess = {
-                    removeUninstalledAddonAtPosition(position)
-                },
-                onError = { addonId, throwable ->
-                    unsupportedAddonsAdapterDelegate.onUninstallError(addonId, throwable)
-                    holder.removeButton.visibility = View.VISIBLE
-                })
+                    onSuccess = {
+                        pendingUninstall = false
+                        removeUninstalledAddon(addon)
+                    },
+                    onError = { addonId, throwable ->
+                        pendingUninstall = false
+                        notifyDataSetChanged()
+                        unsupportedAddonsAdapterDelegate.onUninstallError(addonId, throwable)
+                    })
         }
     }
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal fun removeUninstalledAddonAtPosition(position: Int) {
-        if (position >= unsupportedAddons.size) {
+    internal fun removeUninstalledAddon(addon: Addon) {
+        if (!unsupportedAddons.remove(addon)) {
             return
         }
-
-        val uninstalledAddon = unsupportedAddons[position]
-        unsupportedAddons.remove(uninstalledAddon)
-        notifyItemRemoved(position)
+        // notify all visible view holders to enable removeButton
```

Comment:
nit: would remove comment. really not just notifying because of the remove button here, also to reflect the removed add-on :)